identify_monographs.py

```diff
@@ -0,0 +1,402 @@
+                monograph_form_values.append(form["a"])
+        elif unpunctuated_a in monograph_forms:
+            monograph_form_values.append(form["a"])
+    logger.debug(
+        f"Monograph forms for MMS ID {marc_record['001'].data}: {monograph_form_values}"
+    )
+    return monograph_form_values
+
+
+def _normalize_alma_monograph_form_values(form_values: list) -> list:
+    """Normalizes form values to a standard format."""
+    mapping = {
+        "Theater advertising": "Advertisements",
+        "Animated films": "Animation",
+        "Cartoons": "Animation",
+        "Commercials": "Television commercials",
+        "Documentaries and factual films and video": "Documentary",
+        "Features": "Feature films",
+        "Student films and video": "Student films",
+    }
+    normalized_form_values = []
+    for value in form_values:
+        # first, remove ending period, if present
+        value = value.rstrip(".")
+        if value in mapping:
+            normalized_form_values.append(mapping[value])
+        else:
+            normalized_form_values.append(value)
+    return normalized_form_values
+
+
+def _has_non_monograph_alma_title(marc_record) -> bool:
+    """Given a MARC record, checks whether the record has a title value indicating
+    a non-monograph type."""
+    # if $p or $n is in the title, it's not a monograph
+    marc_245 = marc_record.get_fields("245")
+    for field in marc_245:
+        if field["p"] or field["n"]:
+            return True
+    # if any 250 fields are present, it's not a monograph
+    marc_250 = marc_record.get_fields("250")
+    if marc_250:
+        return True
+    # if any 505 fields are present, it's not a monograph
+    marc_505 = marc_record.get_fields("505")
+    if marc_505:
+        return True
+    return False
+
+
+def _match_records(alma_data: list, fm_data: list) -> list:
+    """Matches Alma and FileMaker records based on MMS ID."""
+    fm_data_dict = {fm_record["inventory_no"]: fm_record for fm_record in fm_data}
+    matched_records = [
+        (alma_record, fm_data_dict[alma_record["Permanent Call Number"]])
+        for alma_record in alma_data
+        if alma_record["Permanent Call Number"] in fm_data_dict
+    ]
+    return matched_records
+
+
+def _get_full_bib_data(matched_data: list, alma_bib_file: str) -> list:
+    """Given a list of tuples of Alma data references and FileMaker data, returns a list of
+    tuples of full Alma bib data and FileMaker data."""
+    alma_bib_data = pymarc.MARCReader(open(alma_bib_file, "rb"))
+    alma_bib_dict = {record["001"].data: record for record in alma_bib_data}
+    full_data = []
+    for alma_data, fm_data in matched_data:
+        alma_record = alma_bib_dict.get(alma_data["MMS Id"])
+        full_data.append((alma_record, fm_data))
+    return full_data
+
+
+def is_monograph(alma_record: Record, fm_record: dict) -> bool:
+    """Determines whether a given record is a monograph."""
```

Comment:
This docstring should be clearer, since the method checks two kinds of "record", and the Alma record is only relevant if the FM record has no production type. 

I think the implementation could be clearer too. The two top-level tests are separate; could this be rewritten as an `if-else`, rather than two (unrelated?) `if`s? Maybe additional helper method(s) which return the monograph-ness of Alma record and FM record, individually? Then this `is_monograph` method is simpler, with Alma-specific and FM-specific logic in those separate helper methods.